### PR TITLE
fix: Solves glitch issue(multiple selected to none) while dropping a building block.

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description
This PR does 2 things: 
1. Extracts some of pasting functionality to its utility functions for readability
2. Includes a new algo for building block pasting to incorporate for specific asks of BBs

Fixes #`Issue Number`  
_or_  
Fixes `Issue URL`
> [!WARNING]  
> _If no issue exists, please create an issue first, and check with the maintainers if the issue is valid._

## Automation

/ok-to-test tags="@tag.Widget, @tag.Sanity"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/9011680217>
> Commit: 1c7bd61f91e3f8ac8d9f4bf8cb71637fd240fa83
> Cypress dashboard url: <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=9011680217&attempt=1" target="_blank">Click here!</a>

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [x] No

### CODE REVIEW

File: app/client/src/sagas/WidgetAdditionSagas.ts [hunk 1]
```diff
@@ -52,7 +52,7 @@ import {
 } from "./WidgetBlueprintSagas";
 import { getPropertiesToUpdate } from "./WidgetOperationSagas";
 import { getWidget, getWidgets } from "./selectors";
-import { addBuildingBlockToCanvasSaga } from "./BuildingBlockAdditionSagas";
+import { addBuildingBlockToCanvasSaga } from "./BuildingBlockSagas/BuildingBlockAdditionSagas";
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/sagas/WidgetAdditionSagas.ts:55]: But it is used in line 511. Why do you think this is unused?

**Review by jacquesikot — APPROVED** [review]

File: app/client/src/sagas/BuildingBlockSagas/BuildingBlockAdditionSagas.ts [hunk 2]
```diff
@@ -0,0 +1,872 @@
...
+  const skeletonWidget: FlattenedWidgetProps = yield select(
+    getWidgetByName,
+    skeletonWidgetName,
+  );
+  yield call(
+    loadBuildingBlocksIntoApplication,
+    addEntityAction.payload,
+    skeletonWidget.widgetId,
+  );
+}
```
[the remark this answers is not in the document]
Comment (rahulbarwal) [reply app/client/src/sagas/BuildingBlockSagas/BuildingBlockAdditionSagas.ts:333]: @jacquesikot **we have to be careful here.** We were lucky that this was only used for logging purposes.

**Review by jacquesikot — APPROVED** [review]

**Comment by marks0351** [comment]
@rahulbarwal @jacquesikot how/when do we plan to implement the proper atomic operations for building blocks instead of the pseudo paste mechanism?

**Comment by rahulbarwal** [comment]
>@rahulbarwal @jacquesikot how/when do we plan to implement the proper atomic operations for building blocks instead of the pseudo paste mechanism?

@marks0351, this PR is a preliminary step in the direction we're considering. We've currently isolated the paste functionality because it necessitates additional logic in the case of BB, and we've eliminated any non-required elements. 

Please note that we're still in the experimental phase, hence the ongoing adjustments. As we gain more insight into DSL and widgets in the upcoming iterations, we'll strategize more streamlined atomic operations. We'll keep you updated on our progress.
cc: @jacquesikot

**Comment by marks0351** [comment]
> > @rahulbarwal @jacquesikot how/when do we plan to implement the proper atomic operations for building blocks instead of the pseudo paste mechanism?
> 
> @marks0351, this PR is a preliminary step in the direction we're considering. We've currently isolated the paste functionality because it necessitates additional logic in the case of BB, and we've eliminated any non-required elements.
> 
> Please note that we're still in the experimental phase, hence the ongoing adjustments. As we gain more insight into DSL and widgets in the upcoming iterations, we'll strategize more streamlined atomic operations. We'll keep you updated on our progress. cc: @jacquesikot

Thanks for clarifying. Do we have tasks created for the issues that we plan to tackle? If not I would recommend to have it so that we don't miss any vulnerable code pieces that we encounter in this current phase.

**Pull request merged**